File: year/2025/day04/task1/main.cpp
/*
Author: Jose A. Romero
Puzzle: Advent of Code (year=2025 ; day=4 ; task=1)
*/

#include <bits/stdc++.h>

#define endl '\n'

using namespace std;

int main() {
    ios_base::sync_with_stdio(0), cin.tie(0), cout.tie(0);

    vector<vector<char>> grid;
    string line;
    while (getline(cin, line)) {
        grid.push_back({});
        for (auto c : line) {
            grid.back().push_back(c);
        }
    }

    int n = grid.size(), m = grid[0].size();
    int ans = 0;

    for (int i = 0 ; i < n ; ++i) {
        for (int j = 0 ; j < m ; ++j) {
            if (grid[i][j] == '@') {
                int cnt = 0;
                for (int di = -1 ; di <= 1 ; ++di) {
                    for (int dj = -1 ; dj <= 1 ; ++dj) {
                        cnt += 0 <= i + di && i + di < n && 0 <= j + dj && j + dj < m && grid[i + di][j + dj] == '@';
                    }
                }
                ans += cnt < 5;  // It should be less than 5 because when di = dj = 0 then it counts itself :)
            }
        }
    }
    cout << ans << endl;
    return 0;
}

File: year/2025/day04/task1/main.py
"""
Author: Jose A. Romero
Puzzle: Advent of Code (year=2025 ; day=4 ; task=1)
"""

import sys


def main() -> None:
    grid = sys.stdin.read().splitlines()

    n, m = len(grid), len(grid[0])
    ans = 0

    for i in range(n):
        for j in range(m):
            if grid[i][j] == "@":
                cnt = 0
                for di in range(-1, 2, 1):
                    for dj in range(-1, 2, 1):
                        cnt += (0 <= i + di < n) and (0 <= j + dj < m) and grid[i + di][j + dj] == "@"

                ans += cnt < 5  # It should be less than 5 because when di = dj = 0 then it counts itself :)
    print(ans)


if __name__ == "__main__":
    main()
